<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1027</width>
    <height>662</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>10000000</width>
    <height>10000000</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Обучение глубоких нейронных сетей</string>
  </property>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>100</y>
     <width>171</width>
     <height>51</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>161</width>
     <height>0</height>
    </size>
   </property>
   <property name="mouseTracking">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Размерность тестовых данных в %</string>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QLabel" name="MSE_label">
   <property name="geometry">
    <rect>
     <x>12</x>
     <y>410</y>
     <width>136</width>
     <height>32</height>
    </rect>
   </property>
   <property name="text">
    <string>Среднеквадратичная ошибка (MSE):</string>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QLabel" name="MSE">
   <property name="geometry">
    <rect>
     <x>162</x>
     <y>420</y>
     <width>84</width>
     <height>18</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>84</width>
     <height>0</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>84</width>
     <height>16777215</height>
    </size>
   </property>
   <property name="frameShape">
    <enum>QFrame::Panel</enum>
   </property>
   <property name="lineWidth">
    <number>0</number>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="textFormat">
    <enum>Qt::AutoText</enum>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="RMSE_label">
   <property name="geometry">
    <rect>
     <x>8</x>
     <y>460</y>
     <width>136</width>
     <height>32</height>
    </rect>
   </property>
   <property name="text">
    <string>Среднеквадратичная ошибка (RMSE):</string>
   </property>
   <property name="textFormat">
    <enum>Qt::PlainText</enum>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QLabel" name="RMSE">
   <property name="geometry">
    <rect>
     <x>152</x>
     <y>460</y>
     <width>84</width>
     <height>18</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>84</width>
     <height>0</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>84</width>
     <height>16777215</height>
    </size>
   </property>
   <property name="frameShape">
    <enum>QFrame::Panel</enum>
   </property>
   <property name="lineWidth">
    <number>0</number>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="MAE_label">
   <property name="geometry">
    <rect>
     <x>12</x>
     <y>500</y>
     <width>141</width>
     <height>71</height>
    </rect>
   </property>
   <property name="text">
    <string>Средняя абсолютная ошибка (MAE)</string>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QLabel" name="MAE">
   <property name="geometry">
    <rect>
     <x>182</x>
     <y>520</y>
     <width>84</width>
     <height>18</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>84</width>
     <height>0</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>84</width>
     <height>16777215</height>
    </size>
   </property>
   <property name="frameShape">
    <enum>QFrame::Panel</enum>
   </property>
   <property name="lineWidth">
    <number>0</number>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="trainingCyclesNumberLabel">
   <property name="geometry">
    <rect>
     <x>14</x>
     <y>243</y>
     <width>131</width>
     <height>48</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>90</width>
     <height>0</height>
    </size>
   </property>
   <property name="text">
    <string>Количество циклов обучения</string>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QSpinBox" name="epochs_SpinBox">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>260</y>
     <width>60</width>
     <height>20</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>60</width>
     <height>20</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>60</width>
     <height>20</height>
    </size>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="maximum">
    <number>1000</number>
   </property>
  </widget>
  <widget class="QLabel" name="activationFunctionLabel">
   <property name="geometry">
    <rect>
     <x>360</x>
     <y>60</y>
     <width>101</width>
     <height>81</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="text">
    <string>Функция активации для первого слоя</string>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QComboBox" name="activation_ComboBox">
   <property name="geometry">
    <rect>
     <x>480</x>
     <y>90</y>
     <width>121</width>
     <height>31</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>60</width>
     <height>20</height>
    </size>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>9</x>
     <y>303</y>
     <width>124</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Скорость обучения</string>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QLabel" name="trainingRateTitle">
   <property name="geometry">
    <rect>
     <x>382</x>
     <y>236</y>
     <width>0</width>
     <height>0</height>
    </rect>
   </property>
   <property name="maximumSize">
    <size>
     <width>0</width>
     <height>16777215</height>
    </size>
   </property>
   <property name="text">
    <string>Скорость обучения</string>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="learningRate_DoubleSpinBox">
   <property name="geometry">
    <rect>
     <x>150</x>
     <y>300</y>
     <width>60</width>
     <height>20</height>
    </rect>
   </property>
   <property name="maximumSize">
    <size>
     <width>60</width>
     <height>20</height>
    </size>
   </property>
   <property name="decimals">
    <number>3</number>
   </property>
   <property name="minimum">
    <double>0.000000000000000</double>
   </property>
   <property name="maximum">
    <double>1.000000000000000</double>
   </property>
   <property name="singleStep">
    <double>0.001000000000000</double>
   </property>
   <property name="value">
    <double>0.001000000000000</double>
   </property>
  </widget>
  <widget class="QLabel" name="trainingMethodLabel">
   <property name="geometry">
    <rect>
     <x>375</x>
     <y>52</y>
     <width>90</width>
     <height>20</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>90</width>
     <height>0</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>90</width>
     <height>16777215</height>
    </size>
   </property>
   <property name="text">
    <string>Метод обучения</string>
   </property>
  </widget>
  <widget class="QComboBox" name="trainingMethod">
   <property name="geometry">
    <rect>
     <x>470</x>
     <y>50</y>
     <width>104</width>
     <height>26</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>ADAM</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>ADADELTA</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>SGD</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>RMSProp</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>RandomForest</string>
    </property>
   </item>
  </widget>
  <widget class="QPushButton" name="pushButton">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>620</y>
     <width>161</width>
     <height>32</height>
    </rect>
   </property>
   <property name="text">
    <string>Сохранить</string>
   </property>
  </widget>
  <widget class="QPushButton" name="openFileButton">
   <property name="geometry">
    <rect>
     <x>310</x>
     <y>10</y>
     <width>341</width>
     <height>32</height>
    </rect>
   </property>
   <property name="text">
    <string>Открыть файл с датасетом</string>
   </property>
  </widget>
  <widget class="QLabel" name="validationSizeLabel">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>140</y>
     <width>181</width>
     <height>48</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>90</width>
     <height>0</height>
    </size>
   </property>
   <property name="mouseTracking">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Размерность валидационных данных в %</string>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QLabel" name="batchSizeLabel">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>330</y>
     <width>161</width>
     <height>64</height>
    </rect>
   </property>
   <property name="text">
    <string>Количество примеров, подаваемых за раз</string>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QSpinBox" name="batchSize_SpinBox">
   <property name="geometry">
    <rect>
     <x>170</x>
     <y>350</y>
     <width>60</width>
     <height>20</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>60</width>
     <height>20</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>60</width>
     <height>20</height>
    </size>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="maximum">
    <number>1000</number>
   </property>
  </widget>
  <widget class="QLabel" name="neuronsNumberFirstLayerLabel">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>240</y>
     <width>191</width>
     <height>48</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>90</width>
     <height>0</height>
    </size>
   </property>
   <property name="text">
    <string>Количество нейронов для первого слоя </string>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QSpinBox" name="neuronsNumberFirstLayer_SpinBox">
   <property name="geometry">
    <rect>
     <x>470</x>
     <y>260</y>
     <width>62</width>
     <height>24</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>60</width>
     <height>20</height>
    </size>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="maximum">
    <number>1000</number>
   </property>
  </widget>
  <widget class="QLabel" name="neuronsNumberSecondLayerLabel">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>280</y>
     <width>151</width>
     <height>80</height>
    </rect>
   </property>
   <property name="text">
    <string>Количество нейронов для второго слоя </string>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QSpinBox" name="neuronsNumberSecondLayer_SpinBox">
   <property name="geometry">
    <rect>
     <x>420</x>
     <y>310</y>
     <width>62</width>
     <height>24</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>60</width>
     <height>20</height>
    </size>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="maximum">
    <number>1000</number>
   </property>
  </widget>
  <widget class="QLabel" name="neuronsNumberOutputLayerLabel">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>340</y>
     <width>161</width>
     <height>64</height>
    </rect>
   </property>
   <property name="text">
    <string>Количество нейронов для выходного слоя </string>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QSpinBox" name="neuronsNumberOutputLayer_SpinBox">
   <property name="geometry">
    <rect>
     <x>430</x>
     <y>360</y>
     <width>62</width>
     <height>24</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>60</width>
     <height>20</height>
    </size>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="maximum">
    <number>1000</number>
   </property>
  </widget>
  <widget class="QPushButton" name="trainingButton">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>620</y>
     <width>141</width>
     <height>32</height>
    </rect>
   </property>
   <property name="text">
    <string>Тренировать</string>
   </property>
  </widget>
  <widget class="QPushButton" name="trainingButton_2">
   <property name="geometry">
    <rect>
     <x>260</x>
     <y>460</y>
     <width>151</width>
     <height>32</height>
    </rect>
   </property>
   <property name="text">
    <string>График потерь</string>
   </property>
  </widget>
  <widget class="QPushButton" name="trainingButton_3">
   <property name="geometry">
    <rect>
     <x>260</x>
     <y>420</y>
     <width>151</width>
     <height>32</height>
    </rect>
   </property>
   <property name="text">
    <string>График точности</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="testSizeValue">
   <property name="geometry">
    <rect>
     <x>240</x>
     <y>120</y>
     <width>60</width>
     <height>20</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>60</width>
     <height>20</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>60</width>
     <height>20</height>
    </size>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="maximum">
    <number>1000</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="testValidateSize">
   <property name="geometry">
    <rect>
     <x>240</x>
     <y>170</y>
     <width>60</width>
     <height>20</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>60</width>
     <height>20</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>60</width>
     <height>20</height>
    </size>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="maximum">
    <number>1000</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>60</y>
     <width>171</width>
     <height>51</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>161</width>
     <height>0</height>
    </size>
   </property>
   <property name="mouseTracking">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Размерность обучающих данных в %</string>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QSpinBox" name="trainDataSize">
   <property name="geometry">
    <rect>
     <x>240</x>
     <y>80</y>
     <width>60</width>
     <height>20</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>60</width>
     <height>20</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>60</width>
     <height>20</height>
    </size>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="maximum">
    <number>1000</number>
   </property>
  </widget>
  <widget class="QTableView" name="tableView">
   <property name="geometry">
    <rect>
     <x>660</x>
     <y>10</y>
     <width>341</width>
     <height>192</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="activationFunctionLabel_2">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>230</y>
     <width>141</width>
     <height>32</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="text">
    <string>Столбец для обучения</string>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QComboBox" name="trainColumn">
   <property name="geometry">
    <rect>
     <x>810</x>
     <y>230</y>
     <width>121</width>
     <height>41</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>60</width>
     <height>20</height>
    </size>
   </property>
  </widget>
  <widget class="QLabel" name="activationFunctionLabel_3">
   <property name="geometry">
    <rect>
     <x>360</x>
     <y>110</y>
     <width>101</width>
     <height>81</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="text">
    <string>Функция активации для второго слоя</string>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QComboBox" name="activation_ComboBox_2">
   <property name="geometry">
    <rect>
     <x>480</x>
     <y>140</y>
     <width>121</width>
     <height>31</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>60</width>
     <height>20</height>
    </size>
   </property>
  </widget>
  <widget class="QComboBox" name="activation_ComboBox_3">
   <property name="geometry">
    <rect>
     <x>480</x>
     <y>190</y>
     <width>121</width>
     <height>31</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>60</width>
     <height>20</height>
    </size>
   </property>
  </widget>
  <widget class="QLabel" name="activationFunctionLabel_4">
   <property name="geometry">
    <rect>
     <x>360</x>
     <y>160</y>
     <width>101</width>
     <height>81</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="text">
    <string>Функция активации для третьего слоя</string>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QLabel" name="activationFunctionLabel_5">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>270</y>
     <width>141</width>
     <height>32</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="text">
    <string>Предсказанные значения</string>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QTableWidget" name="predictedTable">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>300</y>
     <width>256</width>
     <height>192</height>
    </rect>
   </property>
   <column>
    <property name="text">
     <string>Predicted</string>
    </property>
   </column>
  </widget>
  <widget class="QLabel" name="neuronsNumberOutputLayerLabel_2">
   <property name="geometry">
    <rect>
     <x>680</x>
     <y>540</y>
     <width>161</width>
     <height>64</height>
    </rect>
   </property>
   <property name="text">
    <string>Количество дерьевьев</string>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QSpinBox" name="trees">
   <property name="geometry">
    <rect>
     <x>840</x>
     <y>560</y>
     <width>62</width>
     <height>24</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>60</width>
     <height>20</height>
    </size>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="maximum">
    <number>1000</number>
   </property>
  </widget>
  <widget class="QLabel" name="neuronsNumberOutputLayerLabel_3">
   <property name="geometry">
    <rect>
     <x>700</x>
     <y>500</y>
     <width>241</width>
     <height>64</height>
    </rect>
   </property>
   <property name="text">
    <string>Параметры для метода &quot;случайный лес&quot;</string>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QSpinBox" name="max_depth">
   <property name="geometry">
    <rect>
     <x>840</x>
     <y>610</y>
     <width>62</width>
     <height>24</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>60</width>
     <height>20</height>
    </size>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="maximum">
    <number>1000</number>
   </property>
  </widget>
  <widget class="QLabel" name="neuronsNumberOutputLayerLabel_4">
   <property name="geometry">
    <rect>
     <x>680</x>
     <y>590</y>
     <width>161</width>
     <height>64</height>
    </rect>
   </property>
   <property name="text">
    <string>Максимальная глубина</string>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
